new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/TestActions.java

```diff
@@ -519,4 +558,83 @@ public static boolean deployApplication(String appName, String appLocation, Stri
                                           String username, String password, String target) {
     return true;
   }
+
+  // ------------------------ some utility method  -------------------------
+
+  /**
+   * Check running some command and returning expected value
+   *
+   * @param command the command to run
+   * @param expectedValue the expected value should return
+   * @return true on success, false otherwise
+   */
+  public static boolean getExpectedResult(String command, String expectedValue) {
+    logger.info("Running command - \n" + command);
+
+    ExecResult result;
+    try {
+      result = ExecCommand.exec(command);
+      if (result.exitValue() != 0) {
+        logger.info("Command failed with errors " + result.stderr() + "\n" + result.stdout());
+        return false;
+      }
+    } catch (Exception e) {
+      logger.info("Got exception, command failed with errors " + e.getMessage());
+      return false;
+    }
+
+    if (!result.stdout().contains(expectedValue)) {
+      logger.info("Did not get the expected result, expected: " + expectedValue + ", got: " + result.stdout());
+      return false;
+    }
+
+    return true;
+  }
+
+  /**
+   * Check the load balancer functionality, call curl command and check the response contains the managed server info
+   * @param curlCmd curl command to call sample app
+   * @param managedServerNames managed server names to verify the sample app can hit the servers
+   * @param maxIterations max interations to call the curl command
+   * @throws Exception if the web app can not hit one or more managed servers
+   */
+  public static void callWebAppAndCheckForServerNameInResponse(String curlCmd,
+                                                        List<String> managedServerNames,
+                                                        int maxIterations) throws Exception {
+
+    // map with server names and boolean values
+    HashMap<String, Boolean> managedServers = new HashMap<String, Boolean>();
+    managedServerNames.forEach(managedServerName -> {
+      managedServers.put(managedServerName, false);
+    });
+
+    logger.info("Calling webapp " + maxIterations + " times " + curlCmd);
+
+    // check the response contains managed server name
+    int i = 1;
+    while (managedServers.containsValue(false) && (i <= maxIterations)) {
+      logger.info("Iteration {0}/{1} executing command: {2} ", i, maxIterations, curlCmd);
+
+      ExecResult result = ExecCommand.exec(curlCmd, true);
+
+      String response = result.stdout().trim();
+      managedServers.keySet().forEach(key -> {
+        if (response.contains(key)) {
+          managedServers.put(key, true);
+        }
+      });
+
+      i++;
```

Comment:
Looks like a for-loop.  Not sure why you used while-loop.

Reply:
To me, they are the same. I changed it to the for loop anyway.